src/states/elder/mod.rs

```diff
@@ -2030,6 +2036,12 @@ impl Approved for Elder {
         Ok(())
     }
 
+    fn handle_prune(&mut self) -> Result<(), RoutingError> {
+        let _sec_info = NetworkEvent::SectionInfo(self.chain.prune()?);
+        self.finalise_prefix_change()?;
```

Comment:
This calls `Chain::check_and_clean_neighbour_infos` I think, do we want that?

Reply:
I will go to separate it out.